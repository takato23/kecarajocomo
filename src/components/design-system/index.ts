// Base UI Components
export { Button } from './Button';
export type { ButtonProps } from './Button';

export { Input } from './Input';
export type { InputProps } from './Input';

export { Card, CardHeader, CardBody, CardFooter } from './Card';
export type { CardProps, CardHeaderProps, CardBodyProps, CardFooterProps } from './Card';

export { Badge } from './Badge';
export type { BadgeProps } from './Badge';

export { Heading, Text, Code, Link } from './Typography';
export type { HeadingProps, TextProps, CodeProps, LinkProps } from './Typography';

// Specialized Food/Recipe Components
export { RecipeCard } from './RecipeCard';
export type { RecipeCardProps } from './RecipeCard';

export { NutritionDisplay } from './NutritionDisplay';
export type { NutritionDisplayProps, NutritionInfo } from './NutritionDisplay';

// Design System Utilities
export * from './theme';
export * from './icons';