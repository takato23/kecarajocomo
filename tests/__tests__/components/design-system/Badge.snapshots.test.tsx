import React from 'react';
import { render } from '@testing-library/react';
import { Badge } from '@/components/design-system/Badge';

// Mock icons for consistent snapshots
const MockIcon = () => <span>⭐</span>;

describe('Badge Snapshots', () => {
  describe('Badge Variants', () => {
    it('should render default badge', () => {
      const { container } = render(<Badge>Default</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render fresh badge', () => {
      const { container } = render(<Badge variant="fresh">Fresh</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render warm badge', () => {
      const { container } = render(<Badge variant="warm">Warm</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render rich badge', () => {
      const { container } = render(<Badge variant="rich">Rich</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render golden badge', () => {
      const { container } = render(<Badge variant="golden">Golden</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render neutral badge', () => {
      const { container } = render(<Badge variant="neutral">Neutral</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render success badge', () => {
      const { container } = render(<Badge variant="success">Success</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render warning badge', () => {
      const { container } = render(<Badge variant="warning">Warning</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render error badge', () => {
      const { container } = render(<Badge variant="error">Error</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render info badge', () => {
      const { container } = render(<Badge variant="info">Info</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Badge Outline Variants', () => {
    it('should render outline default badge', () => {
      const { container } = render(<Badge outline>Outline Default</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render outline fresh badge', () => {
      const { container } = render(<Badge variant="fresh" outline>Outline Fresh</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render outline success badge', () => {
      const { container } = render(<Badge variant="success" outline>Outline Success</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render outline error badge', () => {
      const { container } = render(<Badge variant="error" outline>Outline Error</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Badge Sizes', () => {
    it('should render small badge', () => {
      const { container } = render(<Badge size="sm">Small</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render medium badge', () => {
      const { container } = render(<Badge size="md">Medium</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render large badge', () => {
      const { container } = render(<Badge size="lg">Large</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Badge Shapes', () => {
    it('should render default rounded badge', () => {
      const { container } = render(<Badge>Default Shape</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render pill badge', () => {
      const { container } = render(<Badge pill>Pill Shape</Badge>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Badge with Icons', () => {
    it('should render badge with left icon', () => {
      const { container } = render(
        <Badge leftIcon={<MockIcon />}>Left Icon</Badge>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render badge with right icon', () => {
      const { container } = render(
        <Badge rightIcon={<MockIcon />}>Right Icon</Badge>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render badge with both icons', () => {
      const { container } = render(
        <Badge leftIcon={<MockIcon />} rightIcon={<MockIcon />}>
          Both Icons
        </Badge>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Badge Removable', () => {
    it('should render removable badge', () => {
      const { container } = render(
        <Badge removable onRemove={() => {}}>Removable</Badge>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render removable badge with left icon', () => {
      const { container } = render(
        <Badge leftIcon={<MockIcon />} removable onRemove={() => {}}>
          Removable with Icon
        </Badge>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render removable pill badge', () => {
      const { container } = render(
        <Badge pill removable onRemove={() => {}}>
          Removable Pill
        </Badge>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Badge Complex Combinations', () => {
    it('should render large fresh pill badge with icon', () => {
      const { container } = render(
        <Badge variant="fresh" size="lg" pill leftIcon={<MockIcon />}>
          Fresh Pill
        </Badge>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render outline warning badge with both icons', () => {
      const { container } = render(
        <Badge 
          variant="warning" 
          outline 
          leftIcon={<MockIcon />} 
          rightIcon={<MockIcon />}
        >
          Warning Badge
        </Badge>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render small removable success badge', () => {
      const { container } = render(
        <Badge 
          variant="success" 
          size="sm" 
          removable 
          onRemove={() => {}}
        >
          Success
        </Badge>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render large outline golden pill badge with left icon', () => {
      const { container } = render(
        <Badge 
          variant="golden" 
          size="lg" 
          outline 
          pill 
          leftIcon={<MockIcon />}
        >
          Golden Pill
        </Badge>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render rich removable badge with left icon', () => {
      const { container } = render(
        <Badge 
          variant="rich" 
          leftIcon={<MockIcon />} 
          removable 
          onRemove={() => {}}
        >
          Rich Removable
        </Badge>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Badge Custom Props', () => {
    it('should render badge with custom className', () => {
      const { container } = render(
        <Badge className="custom-badge">Custom Badge</Badge>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render badge with custom attributes', () => {
      const { container } = render(
        <Badge data-testid="custom-badge" aria-label="Custom Badge">
          Custom Attributes
        </Badge>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Badge Status Examples', () => {
    it('should render status badges collection', () => {
      const { container } = render(
        <div className="flex gap-2">
          <Badge variant="success">Active</Badge>
          <Badge variant="warning">Pending</Badge>
          <Badge variant="error">Inactive</Badge>
          <Badge variant="info">Draft</Badge>
        </div>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render food category badges', () => {
      const { container } = render(
        <div className="flex gap-2">
          <Badge variant="fresh">Fresh</Badge>
          <Badge variant="warm">Warm</Badge>
          <Badge variant="rich">Rich</Badge>
          <Badge variant="golden">Golden</Badge>
        </div>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});