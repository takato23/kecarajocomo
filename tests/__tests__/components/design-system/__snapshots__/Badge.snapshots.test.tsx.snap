// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Snapshots Badge Complex Combinations should render large fresh pill badge with icon 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-full px-3 py-1.5 text-base bg-food-fresh-100 text-food-fresh-800 border border-food-fresh-200 dark:bg-food-fresh-900/30 dark:text-food-fresh-200 dark:border-food-fresh-800"
>
  <span
    class="flex-shrink-0 w-5 h-5"
  >
    <span>
      ⭐
    </span>
  </span>
  Fresh Pill
</span>
`;

exports[`Badge Snapshots Badge Complex Combinations should render large outline golden pill badge with left icon 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-full px-3 py-1.5 text-base bg-transparent border border-food-golden-300 text-food-golden-700 dark:border-food-golden-400 dark:text-food-golden-300"
>
  <span
    class="flex-shrink-0 w-5 h-5"
  >
    <span>
      ⭐
    </span>
  </span>
  Golden Pill
</span>
`;

exports[`Badge Snapshots Badge Complex Combinations should render outline warning badge with both icons 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-transparent border border-warning-300 text-warning-700 dark:border-warning-400 dark:text-warning-300"
>
  <span
    class="flex-shrink-0 w-4 h-4"
  >
    <span>
      ⭐
    </span>
  </span>
  Warning Badge
  <span
    class="flex-shrink-0 w-4 h-4"
  >
    <span>
      ⭐
    </span>
  </span>
</span>
`;

exports[`Badge Snapshots Badge Complex Combinations should render rich removable badge with left icon 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-food-rich-100 text-food-rich-800 border border-food-rich-200 dark:bg-food-rich-900/30 dark:text-food-rich-200 dark:border-food-rich-800"
>
  <span
    class="flex-shrink-0 w-4 h-4"
  >
    <span>
      ⭐
    </span>
  </span>
  Rich Removable
  <button
    class="flex-shrink-0 rounded-full p-0.5 hover:bg-black/10 dark:hover:bg-white/10 focus:outline-none focus:ring-1 focus:ring-current w-4 h-4"
    type="button"
  >
    <svg
      class="w-full h-full"
      fill="none"
      stroke="currentColor"
      stroke-width="2"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 18L18 6M6 6l12 12"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</span>
`;

exports[`Badge Snapshots Badge Complex Combinations should render small removable success badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2 py-0.5 text-xs bg-success-50 text-success-700 border border-success-200 dark:bg-success-900/30 dark:text-success-300 dark:border-success-800"
>
  Success
  <button
    class="flex-shrink-0 rounded-full p-0.5 hover:bg-black/10 dark:hover:bg-white/10 focus:outline-none focus:ring-1 focus:ring-current w-3 h-3"
    type="button"
  >
    <svg
      class="w-full h-full"
      fill="none"
      stroke="currentColor"
      stroke-width="2"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 18L18 6M6 6l12 12"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</span>
`;

exports[`Badge Snapshots Badge Custom Props should render badge with custom attributes 1`] = `
<span
  aria-label="Custom Badge"
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
  data-testid="custom-badge"
>
  Custom Attributes
</span>
`;

exports[`Badge Snapshots Badge Custom Props should render badge with custom className 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700 custom-badge"
>
  Custom Badge
</span>
`;

exports[`Badge Snapshots Badge Outline Variants should render outline default badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-transparent border border-neutral-300 text-neutral-700 dark:border-neutral-600 dark:text-neutral-300"
>
  Outline Default
</span>
`;

exports[`Badge Snapshots Badge Outline Variants should render outline error badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-transparent border border-error-300 text-error-700 dark:border-error-400 dark:text-error-300"
>
  Outline Error
</span>
`;

exports[`Badge Snapshots Badge Outline Variants should render outline fresh badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-transparent border border-food-fresh-300 text-food-fresh-700 dark:border-food-fresh-400 dark:text-food-fresh-300"
>
  Outline Fresh
</span>
`;

exports[`Badge Snapshots Badge Outline Variants should render outline success badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-transparent border border-success-300 text-success-700 dark:border-success-400 dark:text-success-300"
>
  Outline Success
</span>
`;

exports[`Badge Snapshots Badge Removable should render removable badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
>
  Removable
  <button
    class="flex-shrink-0 rounded-full p-0.5 hover:bg-black/10 dark:hover:bg-white/10 focus:outline-none focus:ring-1 focus:ring-current w-4 h-4"
    type="button"
  >
    <svg
      class="w-full h-full"
      fill="none"
      stroke="currentColor"
      stroke-width="2"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 18L18 6M6 6l12 12"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</span>
`;

exports[`Badge Snapshots Badge Removable should render removable badge with left icon 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
>
  <span
    class="flex-shrink-0 w-4 h-4"
  >
    <span>
      ⭐
    </span>
  </span>
  Removable with Icon
  <button
    class="flex-shrink-0 rounded-full p-0.5 hover:bg-black/10 dark:hover:bg-white/10 focus:outline-none focus:ring-1 focus:ring-current w-4 h-4"
    type="button"
  >
    <svg
      class="w-full h-full"
      fill="none"
      stroke="currentColor"
      stroke-width="2"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 18L18 6M6 6l12 12"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</span>
`;

exports[`Badge Snapshots Badge Removable should render removable pill badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-full px-2.5 py-1 text-sm bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
>
  Removable Pill
  <button
    class="flex-shrink-0 rounded-full p-0.5 hover:bg-black/10 dark:hover:bg-white/10 focus:outline-none focus:ring-1 focus:ring-current w-4 h-4"
    type="button"
  >
    <svg
      class="w-full h-full"
      fill="none"
      stroke="currentColor"
      stroke-width="2"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 18L18 6M6 6l12 12"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</span>
`;

exports[`Badge Snapshots Badge Shapes should render default rounded badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
>
  Default Shape
</span>
`;

exports[`Badge Snapshots Badge Shapes should render pill badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-full px-2.5 py-1 text-sm bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
>
  Pill Shape
</span>
`;

exports[`Badge Snapshots Badge Sizes should render large badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-3 py-1.5 text-base bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
>
  Large
</span>
`;

exports[`Badge Snapshots Badge Sizes should render medium badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
>
  Medium
</span>
`;

exports[`Badge Snapshots Badge Sizes should render small badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2 py-0.5 text-xs bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
>
  Small
</span>
`;

exports[`Badge Snapshots Badge Status Examples should render food category badges 1`] = `
<div
  class="flex gap-2"
>
  <span
    class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-food-fresh-100 text-food-fresh-800 border border-food-fresh-200 dark:bg-food-fresh-900/30 dark:text-food-fresh-200 dark:border-food-fresh-800"
  >
    Fresh
  </span>
  <span
    class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-food-warm-100 text-food-warm-800 border border-food-warm-200 dark:bg-food-warm-900/30 dark:text-food-warm-200 dark:border-food-warm-800"
  >
    Warm
  </span>
  <span
    class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-food-rich-100 text-food-rich-800 border border-food-rich-200 dark:bg-food-rich-900/30 dark:text-food-rich-200 dark:border-food-rich-800"
  >
    Rich
  </span>
  <span
    class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-food-golden-100 text-food-golden-800 border border-food-golden-200 dark:bg-food-golden-900/30 dark:text-food-golden-200 dark:border-food-golden-800"
  >
    Golden
  </span>
</div>
`;

exports[`Badge Snapshots Badge Status Examples should render status badges collection 1`] = `
<div
  class="flex gap-2"
>
  <span
    class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-success-50 text-success-700 border border-success-200 dark:bg-success-900/30 dark:text-success-300 dark:border-success-800"
  >
    Active
  </span>
  <span
    class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-warning-50 text-warning-700 border border-warning-200 dark:bg-warning-900/30 dark:text-warning-300 dark:border-warning-800"
  >
    Pending
  </span>
  <span
    class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-error-50 text-error-700 border border-error-200 dark:bg-error-900/30 dark:text-error-300 dark:border-error-800"
  >
    Inactive
  </span>
  <span
    class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-info-50 text-info-700 border border-info-200 dark:bg-info-900/30 dark:text-info-300 dark:border-info-800"
  >
    Draft
  </span>
</div>
`;

exports[`Badge Snapshots Badge Variants should render default badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
>
  Default
</span>
`;

exports[`Badge Snapshots Badge Variants should render error badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-error-50 text-error-700 border border-error-200 dark:bg-error-900/30 dark:text-error-300 dark:border-error-800"
>
  Error
</span>
`;

exports[`Badge Snapshots Badge Variants should render fresh badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-food-fresh-100 text-food-fresh-800 border border-food-fresh-200 dark:bg-food-fresh-900/30 dark:text-food-fresh-200 dark:border-food-fresh-800"
>
  Fresh
</span>
`;

exports[`Badge Snapshots Badge Variants should render golden badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-food-golden-100 text-food-golden-800 border border-food-golden-200 dark:bg-food-golden-900/30 dark:text-food-golden-200 dark:border-food-golden-800"
>
  Golden
</span>
`;

exports[`Badge Snapshots Badge Variants should render info badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-info-50 text-info-700 border border-info-200 dark:bg-info-900/30 dark:text-info-300 dark:border-info-800"
>
  Info
</span>
`;

exports[`Badge Snapshots Badge Variants should render neutral badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
>
  Neutral
</span>
`;

exports[`Badge Snapshots Badge Variants should render rich badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-food-rich-100 text-food-rich-800 border border-food-rich-200 dark:bg-food-rich-900/30 dark:text-food-rich-200 dark:border-food-rich-800"
>
  Rich
</span>
`;

exports[`Badge Snapshots Badge Variants should render success badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-success-50 text-success-700 border border-success-200 dark:bg-success-900/30 dark:text-success-300 dark:border-success-800"
>
  Success
</span>
`;

exports[`Badge Snapshots Badge Variants should render warm badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-food-warm-100 text-food-warm-800 border border-food-warm-200 dark:bg-food-warm-900/30 dark:text-food-warm-200 dark:border-food-warm-800"
>
  Warm
</span>
`;

exports[`Badge Snapshots Badge Variants should render warning badge 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-warning-50 text-warning-700 border border-warning-200 dark:bg-warning-900/30 dark:text-warning-300 dark:border-warning-800"
>
  Warning
</span>
`;

exports[`Badge Snapshots Badge with Icons should render badge with both icons 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
>
  <span
    class="flex-shrink-0 w-4 h-4"
  >
    <span>
      ⭐
    </span>
  </span>
  Both Icons
  <span
    class="flex-shrink-0 w-4 h-4"
  >
    <span>
      ⭐
    </span>
  </span>
</span>
`;

exports[`Badge Snapshots Badge with Icons should render badge with left icon 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
>
  <span
    class="flex-shrink-0 w-4 h-4"
  >
    <span>
      ⭐
    </span>
  </span>
  Left Icon
</span>
`;

exports[`Badge Snapshots Badge with Icons should render badge with right icon 1`] = `
<span
  class="inline-flex items-center gap-1.5 font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-1 rounded-lg px-2.5 py-1 text-sm bg-neutral-100 text-neutral-800 border border-neutral-200 dark:bg-neutral-800 dark:text-neutral-200 dark:border-neutral-700"
>
  Right Icon
  <span
    class="flex-shrink-0 w-4 h-4"
  >
    <span>
      ⭐
    </span>
  </span>
</span>
`;
